pypeit/reduce.py

```diff
@@ -1029,24 +1078,20 @@ def find_objects_pypeline(self, image, std_trace=None,
 
             sobjs.add_sobj(sobjs_slit)
 
-        # If this is a second pass for objfind or this is the first one but skip_second_find is True,
-        # assign slitmask design information to detected objects
+        # If this is a second pass for objfind or this is the first one but skip_second_find is True, extract some
+        # useful info from slitmask design. Matching to the detected objects is done in local_skysub_extract
         if ((self.sobjs_obj is None) and (self.par['reduce']['findobj']['skip_second_find'])) or \
                 (self.sobjs_obj is not None):
             if self.par['reduce']['slitmask']['assign_obj'] and self.slits.maskdef_designtab is not None:
```

Comment:
Are all of these (somewhat ugly) if statements here needed. You created the slit mask object upstream and should know whether assign_obj is true. slits.maskdeg_designtab is an attribute of self.slits. Can you knot simply just exectute

slits.get_maskdef_objpos(...)
slits.get_maskdef_offsset(...) 

without the if/then?